Sources/StreamChatUI/Utils/ImageLoading/ImageLoading.swift

```diff
@@ -69,30 +169,50 @@ public extension ImageLoading {
     ) -> Cancellable? {
         loadImage(
             into: imageView,
-            url: url,
-            imageCDN: imageCDN,
-            placeholder: placeholder,
-            resize: resize,
-            preferredSize: preferredSize,
+            from: url,
+            with: ImageLoaderOptions(
+                resize: preferredSize.map { ImageResize($0) },
+                placeholder: placeholder
+            ),
             completion: completion
         )
     }
-    
+
+    @available(*, deprecated, message: "use loadMultipleImages() instead.")
     func loadImages(
         from urls: [URL],
         placeholders: [UIImage],
         loadThumbnails: Bool = true,
-        thumbnailSize: CGSize = .avatarThumbnailSize,
+        thumbnailSize: CGSize = Components.default.avatarThumbnailSize,
         imageCDN: ImageCDN,
         completion: @escaping (([UIImage]) -> Void)
     ) {
-        loadImages(
-            from: urls,
-            placeholders: placeholders,
-            loadThumbnails: loadThumbnails,
-            thumbnailSize: thumbnailSize,
-            imageCDN: imageCDN,
-            completion: completion
-        )
+        let urlsAndOptions = urls.map { url in
+            (url: url, options: ImageDownloadOptions(resize: .init(thumbnailSize)))
+        }
+
+        downloadMultipleImages(from: urlsAndOptions) { results in
```

Comment:
The reason I didn't do it, is because this logic is kinda weird, and I don't actually understand what it is doing 😄 And since the duplicate is on a deprecated function, I didn't bother to reuse this logic, but probably makes sense, will improve it then 👍